### DISCUSSION TITLE
I want Drift querying, but for my Node.js server

### DISCUSSION THREAD

**Discussion by frankvollebregt** [simolus3/drift#3076] [General]
Hey! This is only tangentially related to Drift itself, but who knows: someone here might have some ideas 🙂

I am working on a Node.js Typescript web server using Nest.js, which talks to a Postgres database (using a database `Pool` from the `pg` package) and provides data through a REST API and in the future also through websockets (hopefully).

What I really like about Drift (well almost everything, but specifically) is how I can use my queries written in SQL, and use the `build_runner` to generate fully typed functions from them, so that I can call them and access the results with IDE hints etc.

Is anyone aware of an NPM package that can achieve something similar? Where I can somehow generate Typescript code based on my SQL queries and database definition, after which I can call my queries as functions and access the result as full Typescript classes?

Any input or pointers are appreciated 😄

**Comment by simolus3** [comment]
Well obviously you should be using dart2js to run your Dart app in node.js /s

Compiled queries in drift are pretty closely linked to other features of drift (e.g. companions for inserts or other placeholders like in `SELECT * FROM tbl ORDER BY $order`), so we probably can't generate typescript code from drift (even though it would sure be interesting).
I've recently became aware of [sqlc](https://sqlc.dev/), which was originally written for Go but also appears to [support typescript generation](https://github.com/sqlc-dev/sqlc-gen-typescript) as a plugin - perhaps that's interesting for you?

**Reply by frankvollebregt** [reply]
Haha, we have actually considered using Dart on the server side in general, but eventually opted for Node.js since it fits better into our existing stack :p

`sqlc` looks promising, I'll definitely take a look! Thank you
